models/agent_model_test.go

```diff
@@ -73,6 +73,28 @@ func TestAgent(t *testing.T) {
 			assert.Equal(t, "mongodb://username:s3cur3%20p%40$$w0r4.@1.2.3.4:12345/", agent.DSN(service, 0, ""))
 		})
 	})
+
+	t.Run("DSN socket", func(t *testing.T) {
+		agent := &Agent{
+			Username: pointer.ToString("username"),
+			Password: pointer.ToString("s3cur3 p@$$w0r4."),
+		}
+		service := &Service{
+			Socket: pointer.ToString("/var/run/mysqld/mysqld.sock"),
+		}
+		for typ, expected := range map[AgentType]string{
+			MySQLdExporterType:          "username:s3cur3 p@$$w0r4.@unix(/var/run/mysqld/mysqld.sock)/database?timeout=1s",
+			ProxySQLExporterType:        "username:s3cur3 p@$$w0r4.@unix(/var/run/mysqld/mysqld.sock)/database?timeout=1s",
```

Comment:
Does Proxy SQL support sockets?

Reply:
Yes, they should.

Reply:
https://github.com/sysown/proxysql/blob/63f09e6889464b61b1a0a6ad36ad082f6c0f5154/etc/proxysql.cnf#L54